src/main/java/core/basesyntax/MyHashMap.java

```diff
@@ -1,19 +1,116 @@
 package core.basesyntax;
 
 public class MyHashMap<K, V> implements MyMap<K, V> {
+    static final int DEFAULT_INITIAL_CAPACITY = 16;
+    static final float DEFAULT_FACTOR = 0.75f;
+    private int currentCapacity;
+    private Node<K, V>[] table;
+    private int threshold;
+    private int size;
+
+    {
```

Comment:
Why did you decide to not use constructor? What is advantages?

Reply:
Actually I  hadn't any, but there we initiate an instance with only one conctructor and I thought, why not). Should I  make a constructor instead!?
What do AI wanted from me? Let's ramble  all his suggestions:
 1. "Using the diamond operator without type parameters is not recommended. Use 'new Node<>' instead of 'new Node". How can I do this If I initiate "table" like new Node <> [ ] will be compile error! 
 2. Where did AI find the class and method implementation begin with empty lines?! I havn't seen any. 
 3.  "Remember to parameterize the Node class: The 'Node' class should be parameterized with <K, V> to ensure type safety." Node class are parameterized "private static class Node<K, V>" !May be I didn't understand AI correctly. Please explain to me.
 4. "Don't use getters and setters in class Node: Direct access to the fields of the 'Node' class should be used instead of getters and setters since 'Node' is a private inner class of 'MyHashMap'." I didn't use any getters and setters in whole class implementation!!!!!!!!!!!!!!
 And all the rest  AI's suggestions as less as are wierd or maybe I didn't get it right then please explain it!

Reply:
You can just use

```
public MyHashMap() {
        this.capacity = DEFAULT_INITIAL_CAPACITY;
        this.table = (Node<K, V>[]) new Node[capacity];
    }
```
    
or 

```
public MyHashMap() {
        table = new Node[DEFAULT_INITIAL_CAPACITY];
        threshold = (int) (table.length * DEFAULT_FACTOR);
        defaultCapacityIncrease = table.length * 2;
    }
```